### PR TITLE
maybe fixes shuttles sometimes hanging the server

### PR DESCRIPTION
**Pull request by Bibby0110**
idk

### CODE REVIEW

File: code/controllers/subsystem/mapping.dm [hunk 1]
```diff
@@ -552,8 +552,9 @@ GLOBAL_LIST_EMPTY(the_station_areas)
 				return reserve
 		//If we didn't return at this point, theres a good chance we ran out of room on the exisiting reserved z levels, so lets try a new one
 		num_of_res_levels += 1
-		var/newReserved = add_new_zlevel("Transit/Reserved [num_of_res_levels]", list(ZTRAIT_RESERVED = TRUE))
-		if(reserve.Reserve(width, height, newReserved))
+		var/datum/space_level/newReserved = add_new_zlevel("Transit/Reserved [num_of_res_levels]", list(ZTRAIT_RESERVED = TRUE))
+		initialize_reserved_level(newReserved.z_value)
+		if(reserve.Reserve(width, height, newReserved.z_value))
```
Comment (ThatLing) [inline code/controllers/subsystem/mapping.dm:557]: Now passes ``newReserved.z_value`` instead of just ``newReserved``

Comment (ThatLing) [reply code/controllers/subsystem/mapping.dm:557]: Proc prototype is ``/datum/turf_reservation/proc/Reserve(width, height, zlevel)``, so is probably correct

Comment (Bibby0110) [reply code/controllers/subsystem/mapping.dm:557]: z_value is defined on newReserved

**Comment by JamieD1** [comment]
Works

**Pull request merged**